### PR TITLE
#166790012 Users Should Be Able To Bookmark Articles

### PR DESCRIPTION
**Pull request by IsaiahRn**
#### What does this PR do?

- Users should be able to bookmark articles for reading later

#### Description of Task to be completed

- Add bookmarking functionality

- Add tests for the endpoint

- Add documentation using swagger

#### How should this be manually tested?

1. Run the following commands in the terminal
```
git clone https://github.com/andela/ah-92explorers-backend.git
cd ah-92explorers-backend
git checkout ft-bookmark-articles-166790012
npm install
npm start
```

2. Run the following requests in Postman using `localhost:3000`:

|  Method  |  Endpoint  |  Task  |
|  --- |  --- |  ---  |
|  `POST`  |  `/api/articles/:slug/bookmark`  |  `Bookmark a specific article`  |
|  `GET`  |  `/api/bookmark`  |  `Fetch all bookmarks of an article`  |
|  `DELETE`  |  `/api/articles/:slug/bookmark`  |  `Delete a bookmarked article`  |

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?
[#166790012](https://www.pivotaltracker.com/story/show/166790012)
#### Screenshots (if appropriate)

- Bookmark article
Screen Shot 2019-07-28 at 13 04 59

- Retrieve bookmarked article
Screen Shot 2019-07-28 at 13 05 25

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I don't have more than 3 major commits on this PR

### CODE REVIEW

**Review by elemanhillary-zz — CHANGES_REQUESTED** [review]
provide screen shot

**Comment by IsaiahRn** [comment]
> provide screen shot

Now, I have updated it

**Review by niyoceles — DISMISSED** [review]

**Review by elemanhillary-zz — DISMISSED** [review]
ready for merge

**Review by kodek-sleuth — DISMISSED** [review]

**Review by niyoceles — DISMISSED** [review]
ready to be merged

**Review by Hervera — APPROVED** [review]
Ready to be merged

**Review by niyoceles — APPROVED** [review]

File: controllers/bookmark.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  */
+  static async bookmark(req, res) {
+    try {
+      const { slug } = req.params;
+      const articleId = await articles.findOne({
+        where: { slug }
+      });
+      if (!articleId) {
+        return res.status(404).json({
+          error: 'Article is not found.'
```
Comment (elemanhillary-zz) [inline controllers/bookmark.js:22]: `article not found`

**Review by elemanhillary-zz — APPROVED** [review]
ready for merge

**Pull request merged**